src/aker_metrics.c

```diff
@@ -19,161 +19,188 @@
 
-	return 1;
+        g_metrics->device_block_count += val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
 }
-int set_aker_metrics(int metrics,int num, ... )
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_window_trans_count()
 {
-	
-	aker_metrics_t * tmp = NULL;
-	tmp = g_metrics;
+	pthread_mutex_lock(&aker_metrics_mut);
 
-	va_list valist;
-	va_start(valist, num);
+	g_metrics->window_trans_count += 1;
 
-	switch(metrics)
-	{
-		case 0:
-			tmp->device_block_count += va_arg(valist, uint32_t);
-			va_end(valist);
-			debug_print("After changing tmp->device_block_count %d\n", tmp->device_block_count);
-			break;
-
-		case 1:
-			tmp->windows_transistion_count += va_arg(valist, uint32_t);
-			va_end(valist);
-			debug_print("After changing tmp->windows_transistion_count %d\n", tmp->windows_transistion_count);
-			break;
-
-		case 2:
-			tmp->schedule_set_count += va_arg(valist, uint32_t);
-			va_end(valist);
-			debug_print("After changing tmp->schedule_set_count %d\n", tmp->schedule_set_count);
-			break;
-
-		case 3:
-			tmp->md5_error_count += va_arg(valist, uint32_t);
-			va_end(valist);
-			debug_print("After changing tmp->md5_error_count %d\n", tmp->md5_error_count);
-			break;
-
-		case 4:
-			tmp->process_start_time = va_arg(valist, time_t);
-			va_end(valist);
-			debug_print("After changing tmp->process_start_time %ld\n", (long)tmp->process_start_time);
-			break;
-
-		case 5:
-			tmp->schedule_enabled = va_arg(valist, int);
-			va_end(valist);
-			debug_print("After changing tmp->schedule_enabled %d\n", tmp->schedule_enabled);
-			break;
-
-		case 6:
-			tmp->timezone = NULL;
-			tmp->timezone = strdup(va_arg(valist, char*));
-			va_end(valist);
-			debug_print("After changing tmp->timezone %d\n", tmp->timezone);
-			break;
-
-		default:
-			debug_error("The mentioned metrics is not found\n");
-			return 0;
-	}
-	return 1;	
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_schedule_set_count()
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->schedule_set_count += 1;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_md5_err_count()
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->md5_err_count += 1;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_process_start_time( time_t val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->process_start_time = val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_schedule_enabled( int val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->schedule_enabled = val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
 }
 
-int stringify_metrics(int flag)
+/* See aker_metrics.h for details. */
+void aker_metric_set_tz( const char *val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->timezone = NULL;
```

Comment:
This may leak memory if g_metrics->timezone was previously set.

```c
    if( g_metrics->timezone ) {
        free(g_metrics->timezone);
    }
    g_metrics->timezone = strdup(val);
```